aten/src/ATen/native/Distance.cpp

```diff
@@ -42,12 +42,36 @@ Tensor cdist(const Tensor& x1, const Tensor& x2, const double p) {
 
   int64_t r1 = x1.size(-2);
   int64_t r2 = x2.size(-2);
-  Tensor result = at::empty({r1, r2}, x1.options());
+  auto dim1 = x1.dim();
+  auto dim2 = x2.dim();
+
+  IntArrayRef batch_tensor1(x1.sizes().data(), std::max<int64_t>(dim1 - 2, 0));
```

Comment:
Can you add some comments to this bunch of code to explain what you are doing

Reply:
Added

Reply:
Fixed